pkg/analysis_server/test/services/correction/fix_test.dart

```diff
@@ -5915,6 +5915,92 @@ main() {
 ''');
   }
 
+  test_lint_replaceWithCollectionLiteral_linked_hash_map_with_generic() async {
+    String src = '''
+import 'dart:collection';
+
+final a = /*LINT*/new LinkedHashMap<int,int>();
```

Comment:
The tests are using a mock sdk, defined in `pkg/analysis_server/test/mock_sdk.dart`. The definition of dart:collection just needs to be extended with the minimal subset of the class LinkedHashMap needed to pass the test.

That said, the test has uncovered a bug in the lint code that needs to be fixed: it throws an exception when the type of the class is unknown because it isn't define anywhere.